### DISCUSSION TITLE
Textbox horizontal scaling for selected text in fabric js

### DISCUSSION THREAD

**Discussion by ubhoopathy** [fabricjs/fabric.js#9655] [Q&A] [answered]
how to textbox horizontal scaling for selected text in fabric js

**Comment by ShaMan123** [comment, marked as answer]
you need to replace the controls
text has resize width controls so replace them with the default ones

`text.controls = createObjectDefaultControls()` or whatever

https://github.com/fabricjs/fabric.js/blob/81d387dd1df8572c58cb48f84f5ce1858152397d/src/controls/commonControls.ts#L13-L106

**Comment by ubhoopathy** [comment]
I need this text only horizontal scaling in IText or textbox fabric js
![Screenshot 2024-02-06 164049](https://github.com/fabricjs/fabric.js/assets/149072388/129d291a-03be-4ab7-acaa-302a140486aa)
